src/tools/sfdk/sdkmanager.cpp

```diff
@@ -1348,6 +1348,9 @@ bool SdkManager::mapEnginePaths(QString *program, QStringList *arguments, QStrin
         {cleanSharedSrc, Constants::BUILD_ENGINE_SHARED_SRC_MOUNT_POINT, caseInsensitiveOnWindows}
 #endif
     };
+    *program = QDir::fromNativeSeparators(*program);
+    for (QString& argument : *arguments)
+         argument = QDir::fromNativeSeparators(argument);
```

Comment:
Nice but I am afraid it would be too generic - backslashes may be used in arguments for more purposes, not just as a directory separator.

Reply:
> Nice but I am afraid it would be too generic - backslashes may be used in arguments for more purposes, not just as a directory separator.

That's what I'm afraid myself as well. But can we really know when the backslash is a directory separator and when it's not?

Reply:
It looks like merssh used simply checking if there exists a file for the argument. I was afraid this would not work with paramaters like "--foo=C:\bar", but I guess we don't use parameters like that?

Reply:
merssh have quite limited set of use cases. sfdk has those 'exec' and 'build-shell' commands, so we do not know what to expect. Maybe we could do something like

```
QRegularExpression re(QRegularExpression::escape(sharedHome) + "([^:;,'\"]*)");
arg.replace(re, sharedHomeMountPoint + "\\1");
```

i.e. replace everything that looks like a file path under sharedHome, up to a well known separators/terminators like `[:;,'"]`.

Reply:
Now it's done in a similar fashion as was in merssh

Reply:
> merssh have quite limited set of use cases. sfdk has those 'exec' and 'build-shell' commands, so we do not know what to expect. Maybe we could do something like
> 
> ```
> QRegularExpression re(QRegularExpression::escape(sharedHome) + "([^:;,'\"]*)");
> arg.replace(re, sharedHomeMountPoint + "\\1");
> ```
> 
> i.e. replace everything that looks like a file path under sharedHome, up to a well known separators/terminators like `[:;,'"]`.

Sounds like a good idea. I can do it, but I won't have time until tomorrow (I find it hard to write bug free code while in a meeting...)